sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnectionCache.java

```diff
@@ -0,0 +1,268 @@
+                    .log("Waiting to connect and become active.");
+
+                return c.connectAndAwaitToActive()
+                    .doOnCancel(() -> {
+                        if (!c.isDisposed()) {
+                            c.closeAsync(closeSignal("The connection request was canceled while waiting to active."))
+                                .subscribe();
+                        }
+                    });
+            })
+            .retryWhen(retryWhenSpec(retryPolicy))
+            .<T>handle((c, sink) -> {
+                @SuppressWarnings("unchecked")
+                final T connection = (T) c;
+                final boolean terminated;
+                synchronized (lock) {
+                    terminated = this.terminated;
+                    currentConnection = connection;
+                }
+                if (terminated) {
+                    connection.closeAsync(closeSignal("Connection recovery support is terminated.")).subscribe();
+                    sink.error(TERMINATED_ERROR);
+                } else {
+                    logger.atInfo()
+                        .addKeyValue(CONNECTION_ID_KEY, c.getId())
+                        .log("Emitting the new active connection.");
+                    sink.next(connection);
+                }
+            }).cacheInvalidateIf(c -> {
+                if (c.isDisposed()) {
+                    logger.atInfo().addKeyValue(CONNECTION_ID_KEY, c.getId())
+                        .log("The connection is closed, requesting a new connection.");
+                    return true;
+                } else {
+                    // Emit cached connection.
+                    return false;
+                }
+            });
+    }
+
+    /**
+     * Get the Mono that, when subscribed, emits the cached connection if it is active or creates and
+     * emits a new connection if the cache is empty or the current cached connection is in closed state.
+     *
+     * @return a Mono that emits active connection.
+     */
+    public Mono<T> get() {
+        return createOrGetCachedConnection;
+    }
+
+    /**
+     * Get the connection fully qualified namespace of the remote broker/resource.
+     *
+     * @return the connection fully qualified namespace.
+     */
+    public String getFullyQualifiedNamespace() {
+        return fullyQualifiedNamespace;
+    }
+
+    /**
+     * Get the relative path to the entity under the fully qualified namespace to which the connection established to.
+     *
+     * @return the entity path.
+     */
+    public String getEntityPath() {
+        return entityPath;
+    }
+
+    /**
+     * Get the retry option object describing the retry parameters driving the connection recovery.
+     *
+     * @return The options for connection recovery retries.
+     */
+    public AmqpRetryOptions getRetryOptions() {
+        return retryOptions;
+    }
+
+    public boolean isCurrentConnectionClosed() {
```

Comment:
It would be helpful to move this comment into the Javadoc so we don't inadvertently use it. (ie. you only see full source code with comments when you download it.... which might not be the case in dependent libs)

Reply:
**[DONE]** (promoted to JavaDoc)